### PR TITLE
Zoomed view

### PR DESCRIPTION
**Pull request by daanlouter**
Apr-04-2019 18-09-11

Will double check and review this tomorrow. Only thing I can think of now is that we should check if the steps to show setting should never exceed total data. Also would be nice to update y scale based on current position

### CODE REVIEW

**Comment by timbrock** [comment]
🏇🏻

**Comment by theduncanclark** [comment]
So excited about this!

> Also would be nice to update y scale based on current position

Yeah, I really think we should do this. Especially for non-ranks mode it will make it both easier to read and much more magical.

We should also add number interpolation even more magic, rename the template "Line chart race" and try to make it go insane on social. We'll need some cracking sample data!

**Comment by daanlouter** [comment]
That's really cool! 😃 A few comments

- When using zoomed y axis, it sometimes starts at -1 even though data doesn't go below 0
- I find the zoomed y axis sometimes quite difficult to read and a bit heavy on the eye. Though it's super useful in some cases, maybe we should have it not as the default?
- Should we cut off the chart at a more sensible location? Think it can look a bit messy when it overlaps at the top.

File: template.yml [hunk 1]
```diff
@@ -64,23 +74,11 @@ settings:
...
-  - property: zoom_enabled
-    name: View
-    type: boolean
-    style: buttons
-    choices:
-      - - Wide
-        - false
-      - - Zoom
-        - true
   - property: zoom_steps_to_show
```
Comment (daanlouter) [inline template.yml]: The steps to show feels a bit weird in this location

**Comment by theduncanclark** [comment]
> When using zoomed y axis, it sometimes starts at -1 even though data doesn't go below 0

Yeah, good point. Currently it just stretches the axis by 20% either size but maybe we should limit that at zero at the bottom unless there's a minus number?

> I find the zoomed y axis sometimes quite difficult to read and a bit heavy on the eye. Though it's super useful in some cases, maybe we should have it not as the default?

I don't feel strongly. A blueprint each way :-)

> Should we cut off the chart at a more sensible location? Think it can look a bit messy when it overlaps at the top.

Yeah, it's a bit sad that it can't continue up behind the header. Where would be more sensible? At the plot? I wonder if that would look better or worse…

File: template.yml (hunk 1 above)
Comment (theduncanclark) [reply template.yml]: Ah, good point, will fix. I didn't mean to leave them there!

Comment (theduncanclark) [reply template.yml]: Ah, good point I didn't mean to leave them there!

**Comment by daanlouter** [comment]
I think this is ready to go? I'll do some more browser testing.

**Comment by daanlouter** [comment]
Actually, looking through the code, I think I'd quite like @timbrock or @theduncanclark  to have another look at this. I couldn't find any controversial things, but then again I'm also the one who wrote the code :-) I tested on Edge, IE, firefox and iOS and seems to work.

**Comment by theduncanclark** [comment]
> Actually, looking through the code, I think I'd quite like @timbrock or @theduncanclark to have another look at this. I couldn't find any controversial things, but then again I'm also the one who wrote the code :-) I tested on Edge, IE, firefox and iOS and seems to work.

Probably makes sense for it to be @timbrock both to avoid me holding it up and because I wrote quite a bit of the code, either pairing with @daanlouter or separately (dynamic y axis was quite a big change!).

**Comment by timbrock** [comment]
I don't understand the "Visible steps" setting. Larger numbers mean the races is squeezed more towards the left handside. The description doesn't help.
Screenshot 2019-04-16 at 16 25 46
Screenshot 2019-04-16 at 16 26 12

**Comment by theduncanclark** [comment]
I've hopefully dealt with @timbrock's points in the two last commits (a44c41c and d50a399).

**Comment by daanlouter** [comment]
Hmm, this seemed to work for me, but it behaved a bit odd when clicking replay:
Apr-18-2019 12-36-44

Have pushed something that hopefully fixes it and simplifies the code a bit.

**Comment by timbrock** [comment]
Tiny tweak: can we swap the Zoomed and Show all buttons so that the Visible stages box is side-by-side with the Zoomed button?

**Comment by daanlouter** [comment]
Have merged this. @theduncanclark would be great if you could take a quick look at https://github.com/kiln/template-horserace/pull/136/commits/873f145f51aeb0076f87d75307d91357a31d5f78 (2 line change), since I changed your original logic there.
